### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by himself65**

### CODE REVIEW

File: types/interface.ts [hunk 1]
```diff
@@ -1,6 +1,57 @@
...
+    [CoseKey.crv]: number
+    [CoseKey.x]: string
+    [CoseKey.y]: string
+}
+
+export interface EcCosePrivateKey extends EcCosePublicKey {
+    [CoseKey.d]: string
+}
+
+export interface PublicKeyCredential extends Credential {
```
Comment (himself65) [inline types/interface.ts:47]: we use our `PublicKeyCredential` because there are some standard incompatibility in `tslib`

File: test/util.ts [hunk 2]
```diff
@@ -68,3 +73,62 @@ export function hex2arrayBuffer(data: string): ArrayBuffer {
...
+            key: {
+                x,
+                y,
+                crv,
+                kty,
+                alg,
+            },
+        })
+
+        // todo: refactor this to node implementation
```
Comment (himself65) [inline test/util.ts:119]: I haven't learned how to use `verify` in nodejs, the nodejs API is too complex and unfriendly docs

Comment (Jack-Works) [reply test/util.ts:119]: this is webcrypto api, maybe MDN doc helps

Comment (himself65) [reply test/util.ts:119]: Yes, but i want to use this api https://nodejs.org/dist/latest-v16.x/docs/api/crypto.html#crypto_crypto_verify_algorithm_data_key_signature_callback

**Review by Jack-Works — APPROVED** [review]

**Pull request merged**